File: hw3/Dos_On_Hash_Table/code6-1.py
file = open("input6-1.txt", 'w')
totle_number = 50000

bucket_list = [11, 23, 47, 97, 199, 409, 823, 1741, 3739, 7517, 15173, 30727, 62233]
bucket_index = 0
file.write('%d\n' %totle_number)

index = 1
for number in range(1, totle_number):
    print number
    bucket_size = bucket_list[bucket_index]
    file.write('%d\n' %(index * bucket_size))
    index = index + 1
    if number == bucket_size:
        bucket_index = bucket_index + 1
        index = 1
file.close()

File: hw3/Dos_On_Hash_Table/code6-1.cpp
#include <iostream>
#include <unordered_map>
#include <fstream>
using namespace std;

int main()
{
    ofstream output_file("input6-1.txt");
    unordered_map<int, int> ht;

    int bucket_size = ht.bucket_count();
    int total_num = 50000;

    output_file << total_num << "\n";
    int num = 0;
    int i = 1;
    while(true)
    {
        int value = i * ht.bucket_count();
        ht[value] = i;
        int nbuckets = ht.bucket_count();
        output_file << value << "\n";
        num++;
        i++;
        if(bucket_size != ht.bucket_count()){
            bucket_size = ht.bucket_count();
            i = 1;
        }
        if(num > total_num)
        {
            break;
        }

    }
    output_file.close();
    return 0;
}
